src/main/webapp/common.js

```diff
@@ -109,9 +110,81 @@ async function displayCurrentAddress() {
 }
 
 /**
- * Method that builds for every page with a "menu-container div" a navigable menu.
+ * If checkboxId element is checked, the function will update the address box addressId to user's
+ * current address so that user can check its location accuracy.
+ */
+function updateAddressFromGeolocation(checkboxId, addressId) {
+  addressBox = document.getElementById(addressId);
+  if (document.getElementById(checkboxId).checked) {
+    // update the address box to show the address found using Geolocation
+    if (navigator.geolocation) {
+        // if location was retrieved successfully, call setLatLng, else alert in geolocationError
+        getLocation().then((position) => {
+          setLatLng(position);
+          getAddressFromLatLng(addressId)
+        }).catch((error) => geolocationError(error));
+    } else {
+      // clear the address box if location is not supported 
+      addressBox.value = ""; 
+      alert("Geolocation is not supported by this browser!");
+    }
+  }
+}
```

Comment:
Could you break this out into two functions - the first one should handle getting the DOM bits and pieces (addressbox and checkbox etc), and then that function should call the second one that handles the `getLocation()` call.

This way you can then write a unit test for the second function :-)